Src/Plugins/Tests.Security/EndToEndTests.cs

```diff
@@ -46,6 +48,30 @@ protected EndToEndTests(ITestOutputHelper outputHelper) : base(outputHelper)
         {
             var inputFiles = parameter as List<string>;
             var results = new Dictionary<string, string>();
+
+            results = Debugger.IsAttached
```

Comment:
>Debugger [](http://example.com/codeflow?start=22&length=8)

In practice, debugging the end-to-end tests when multithreaded is very difficult. The common scenario is to debug one or a few files and it is hard to step through a discrete repro w/many other threads traveling the same breakpoints. With this change, we will be single-threaded when debugging. See code comment for remaining thoughts.